buildSrc/src/main/java/org/opensearch/gradle/zipplugin/ZipPublish.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.gradle.zipplugin;
+
+import java.util.*;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.publish.PublishingExtension;
+import org.gradle.api.publish.maven.MavenPublication;
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin;
+import java.nio.file.Path;
+import org.gradle.api.Task;
+
+public class ZipPublish implements Plugin<Project> {
+    private Project project;
+
+    public final static String EXTENSION_NAME = "zipmavensettings";
+    public final static String PUBLICATION_NAME = "mavenzip";
+    public final static String STAGING_REPO = "zipstaging";
+    public final static String MAVEN_ZIP_PUBLISH_POM_TASK = "generatePomFileForMavenzipPublication";
+    public final static String LOCALMAVEN = "publishToMavenLocal";
+    public final static String LOCAL_STAGING_REPO_PATH = "/build/local-staging-repo";
+    public String zipDistributionLocation = "/build/distributions/";
+
+    private void configMaven(Project project) {
+        final Path buildDirectory = project.getRootDir().toPath();
+        project.getPluginManager().apply(MavenPublishPlugin.class);
+        project.getExtensions().configure(PublishingExtension.class, publishing -> {
```

Comment:
Should we check that the project is actually `PluginBuildPlugin`?

Reply:
@reta I have to plans add this in readme doc as prereq to make sure the zip is built with bundlePlugin task, since this zippublish logic is executed during plugin apply, the task will created and saved as `publishMavenzipPublicationToZipstagingRepository `, an error occurs only during the task runtime, when called in buildscript as `./gradlew publishMavenzipPublicationToZipstagingRepository -Dopensearch.version=$VERSION -Dbuild.snapshot=$SNAPSHOT -Dbuild.version_qualifier=$QUALIFIER`, so adding logic in code to check if the zip is being built or not with bundlePlugin task, could show as error log the entire project runtime and not at the task execution, as apply() methods are called first before all the tasks for the project are executed.

Reply:
That's fine, thanks, as far as we could explain the error clearly, that this publishing task is only available for plugin distribution.

Reply:
Ya I will also add the clear readme doc, that says this plugin publishes zip that is the output of bundlePlugin task `mavenZip.artifact(project.getTasks().named("bundlePlugin"));`